html/captive-portal/lib/captiveportal/PacketFence/Controller/TLSProfile.pm

```diff
@@ -121,23 +131,29 @@ sub cert_process : Private {
     $c->forward( 'CaptivePortal' => 'endPortalSession' );
 }
 
-=head2 validate_form
+=head2 process_form
 
-Validate informations input by the user
+Process form information inputed by the user
 
 =cut
 
 sub process_form : Private {
     my ($self, $c) = @_;
     my $logger = $c->log;
+
     my $portalSession = $c->portalSession;
     my $mac    = $portalSession->clientMac;
     my $passwd = $c->request->param('certificate_pwd');
+    my $provisioner = $c->profile->findProvisioner($mac);
+    my $pki_provider = $provisioner->getPkiProvider();
+    my $pki_provider_name = ref($pki_provider);
+    $pki_provider_name =~ s#^.*:##;
```

Comment:
Getting the pki_provider_name should be either move to function or an attribute.